### PR TITLE
Cleanup PartialEq docs.

### PR DESCRIPTION
**Pull request by frewsxcv**
- Cleanup the `impl PartialEq<BookFormat> for Book` implementation
- Implement `impl PartialEq<Book> for BookFormat` so it’s symmetric
  - Fixes https://github.com/rust-lang/rust/issues/53844.
- Removes the last example since it appears to be redundant with the
  previous two examples.

### CODE REVIEW

**Comment by frewsxcv** [comment]
> A side-effect of the changes here is that there's now no example that uses `match`, which feels like an important teaching opportunity wrt. `PartialEq`.. Could we perhaps work that in in some other way?

In my opinion, this doesn't seem like the right place to teach people about language features like `match`, that seems better suited for TRPL or other introductory learning resources. 

> Also @frewsxcv, note that you cannot actually assign a team so don't use r? for that; just cc the team instead.

Oh I know, I'm just asking someone from the docs team for a review.

**Comment by Centril** [comment]
> In my opinion, this doesn't seem like the right place to teach people about language features like `match`, that seems better suited for TRPL or other introductory learning resources.

OK, you might be right, seems like a judgement call for Steve to make. ;)

> Oh I know, I'm just asking someone from the docs team for a review.

Right, but when you r? a team, rust-highfive just gets unhappy and the PR then has no assignee... =P

File: src/libcore/cmp.rs [hunk 1]
```diff
@@ -96,6 +96,7 @@ use self::Ordering::*;
 /// For example, let's tweak our previous code a bit:
 ///
 /// ```
+/// #[derive(PartialEq)]
```
Comment (QuietMisdreavus) [inline src/libcore/cmp.rs]: While we're explaining everything, i wonder if we can include a comment on this line like "`// Implements <BookFormat> == <BookFormat> comparisons`", just to hammer it home.

File: src/libcore/cmp.rs [hunk 2]
```diff
@@ -107,67 +108,29 @@ use self::Ordering::*;
...
-///     Hardback,
-///     Ebook,
-/// }
-///
-/// struct Book {
-///     isbn: i32,
-///     format: BookFormat,
-/// }
-///
-/// impl PartialEq<BookFormat> for Book {
```
Comment (QuietMisdreavus) [inline src/libcore/cmp.rs]: I think the purpose of the sample you removed was to show off that you can `impl PartialEq for Book` multiple times by changing the type parameter, which allows you to do `<Book> == <Book>` *and* `<Book> == <BookFormat>`. If i remember right, we had a question about that a while back, which is why this was added.

**Review by QuietMisdreavus — COMMENTED** [review]
I like adding the symmetric PartialEq impls, but i'm wary about removing the example you took out.

File: src/libcore/cmp.rs (hunk 1 above)
Comment (frewsxcv) [reply src/libcore/cmp.rs]: done in https://github.com/rust-lang/rust/pull/57357/commits/423a5bb5c437c2ce81d4150490d0cccfdbc33a4c

File: src/libcore/cmp.rs (hunk 2 above)
Comment (frewsxcv) [reply src/libcore/cmp.rs]: fair enough. readded in https://github.com/rust-lang/rust/pull/57357/commits/1445a065ded1bc7e73e7a358f9bc971e908e9181

File: src/libcore/cmp.rs [hunk 3]
```diff
@@ -145,12 +150,13 @@ use self::Ordering::*;
...
-///            (&BookFormat::Paperback, &BookFormat::Paperback) => true,
-///            (&BookFormat::Hardback,  &BookFormat::Hardback)  => true,
-///            (&BookFormat::Ebook,     &BookFormat::Ebook)     => true,
-///            (_, _) => false,
-///         }
+///         self.format == *other
+///     }
+/// }
+///
+/// impl PartialEq<Book> for BookFormat {
```
Comment (QuietMisdreavus) [inline src/libcore/cmp.rs]: Looks like this impl isn't needed in this example any more?

Comment (frewsxcv) [reply src/libcore/cmp.rs]: the impetus of this pull request is that these equality implementations weren't symmetric, even though they should be: https://github.com/rust-lang/rust/issues/53844. does that make sense?

Comment (QuietMisdreavus) [reply src/libcore/cmp.rs]: Right, but this impl isn't being used in this example. This is the one that shows off that you can implement `PartialEq<Thing>` for multiple `Thing`s, and the assertions at the end don't use this specific formation. (You've already edited the previous example to show the asymmetric nature of PartialEq.)

Comment (frewsxcv) [reply src/libcore/cmp.rs]: fair enough. addressed in https://github.com/rust-lang/rust/pull/57357/commits/32b28340b20ea8b45d3e663dc81bb9092ecdeeb1

**Review by QuietMisdreavus — APPROVED** [review]
Looks good, thanks!

**Pull request merged**